packages/-ember-data/tests/integration/store-test.js

```diff
@@ -175,14 +175,12 @@ module('integration/store - destroy', function (hooks) {
 
     next();
 
-    await nextPromise;
+    const result = await nextPromise;
 
     // ensure we allow the internal store promises
     // to flush, potentially pushing data into the store
     await settled();
-    assert.ok(true, 'we made it to the end');
-    await requestPromise;
-    assert.ok(false, 'we should never make it here');
+    assert.equal(result.data.type, 'car', 'we made it to the end');
   });
```

Comment:
This test is now failing on the main branch.  I rewrote it a little bit but the main issue was `await requestPromise`.  On first glance, I can't tell what is going wrong here.

Reply:
Since this was failing on 3.20 (last 3.20 release was https://github.com/emberjs/ember.js/tree/v3.20.7) something has "changed" since the last PRs a week ago.  In any case, I distilled the test down a bit.

Reply:
Ha, interesting, it seems like we are asserting that the promise never finishes, it makes sense it would hang the test. We can rewrite this as `.then` and not hang on it, or just rely on the `store.push`

Reply:
Fixed with `.then`!  Thank you!